src/main/scala/api/sampling/proposals/NonRigidIcpProposal.scala

```diff
@@ -70,10 +69,19 @@ case class NonRigidIcpProposal(
 
 
   override def logTransitionProbability(from: ModelFittingParameters, to: ModelFittingParameters): Double = {
-    val posterior = cashedPosterior(from)
-    val compensatedTo = to.copy(shapeParameters = ShapeParameters(from.shapeParameters.parameters + (to.shapeParameters.parameters - from.shapeParameters.parameters) / stepLength))
-    val pdf = posterior.logpdf(compensatedTo.shapeParameters.parameters)
-    pdf
+    if (to.copy(shapeParameters = from.shapeParameters).allParameters != from.allParameters) {
```

Comment:
Is there a problem when directly comparing the case classes ModelFittingParameters to each other? Or why do you use .allParameters?

Reply:
The ModelFittingParamater case class also has a "generatedBy" string in the constructor parameters which might be different. With .allParamaters we are sure only to compare the (shape, translation, rotation).